### PR TITLE
D8AGE-362: Use github actions for tests.

### PR DESCRIPTION
**Pull request by intelektron**
- Add ci.yml with PHPUnit and PHPCS
- Bump some Composer dependencies to test the minimal version requirements
- Verified that git hooks are up and running

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
   "require-dev": {
     "guzzlehttp/guzzle": "^7",
     "guzzlehttp/psr7": "^2",
-    "openeuropa/code-review": "^2.0.0-alpha6",
```
Comment (intelektron) [inline composer.json:21]: I changed this to 2.0, as the alpha caused random issues with `composer update --prefer-lowest`. I hope this doesn't break any backwards compatibility,

File: composer.json [hunk 2]
```diff
@@ -18,8 +18,8 @@
...
-    "phpunit/phpunit": "^10"
+    "openeuropa/code-review": "^2.0",
+    "phpunit/phpunit": "^10.4"
```
Comment (intelektron) [inline composer.json:22]: If we want to use `Generator.php`, it exists in this form since `v10.4`.

File: composer.json [hunk 3]
```diff
@@ -36,7 +36,8 @@
     "allow-plugins": {
       "phpro/grumphp": true,
       "dealerdirect/phpcodesniffer-composer-installer": true,
-      "php-http/discovery": true
+      "php-http/discovery": true,
+      "cweagans/composer-patches": true
```
Comment (intelektron) [inline composer.json:45]: Needed for `composer update --prefer-lowest`, as some older dependency pulls patches

File: composer.json (hunk 1 above)
Comment (drishu) [reply composer.json:21]: it's ok we will remove it anyway

File: composer.json (hunk 2 above)
Comment (drishu) [reply composer.json:22]: Lets add a section in composer explaining why we use this specific version, ex https://github.com/openeuropa/oe_whitelabel/blob/1.x/composer.json#L104

File: composer.json (hunk 3 above)
Comment (drishu) [reply composer.json:45]: yes, its the openeuropa/codereview, add a note in the readme to remove this as soon as we drop it

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -14,6 +14,9 @@ Use [Composer](https://getcomposer.org/) to install the package:
 $ composer require openeuropa/cdt-client
 ```
 
+***Note**: `cweagans/composer-patches` should be removed from the allowed plugin list after
```
Comment (drishu) [inline README.md]: I wasnt clear enough, I means in the readme section of composer (this doesn't concern the reader of this file), but it's ok, we remove in next PR

**Review by drishu — APPROVED** [review]

**Pull request merged**